enterprise/cmd/frontend/internal/registry/http_api.go

```diff
@@ -31,13 +32,14 @@ var (
 		if err != nil {
 			return nil, err
 		}
-		xs := make([]*registry.Extension, 0, len(vs))
-		for _, v := range vs {
-			x, err := toRegistryAPIExtension(ctx, v)
-			if err != nil {
-				continue
-			}
 
+		xs, err := toRegistryAPIExtensionBatch(ctx, vs)
+		if err != nil {
+			return nil, err
+		}
+
+		ys := make([]*registry.Extension, 0, len(vs))
```

Comment:
```suggestion
		var ys []*registry.Extension
```

A good rule of thumb is to only preallocate when we know the exact length (or there's some other demonstrated perf issue).

Reply:
We know the exact length unless there's malformed data in this case. The jsonc.Unmarshal is really an edge-case check that used to crash clients.